pkg/system/phase4_configuring.go

```diff
@@ -749,3 +833,166 @@ func (r *Reconciler) ReconcileDeploymentEndpointStatus() error {
 
 	return nil
 }
+
```

Comment:
looks like there are quite a lot of azure specific functions. maybe consider moving those to a different file (`azure_utils` or something) in `system` package